ext/curl/curl_file.c

```diff
@@ -117,6 +156,22 @@ ZEND_METHOD(CURLFile, getPostFilename)
 }
 /* }}} */
 
+/* {{{ proto void CURLFile::setFilename(string $name)
+   Set file name */
+ZEND_METHOD(CURLFile, setFilename)
+{
+	curlfile_set_property("name", INTERNAL_FUNCTION_PARAM_PASSTHRU);
+}
+/* }}} */
+
+/* {{{ proto void CURLFile::setBuffer(string $buffer)
+   Set buffer */
+ZEND_METHOD(CURLFile, setBuffer)
+{
+	curlfile_set_property("buffer", INTERNAL_FUNCTION_PARAM_PASSTHRU);
+}
+/* }}} */
```

Comment:
Yes, I fully agree with you! Immutable objects will be perfect for CURLFile. But it this greatly breaks backward compatibility.
Developers can do all this things: 
`curl_buffer_file_create (...)`
`new CURLFile(...)`
`$CURLFile->setFilename = '...'`
`$CURLFile->name=...` ( it's public! )

So if we try to make CURLFile immutable object we required RFC, and many chance that it can't be approved by community. 
For not popular small feature to send file from string we will broke BC. 
I tried to change as less as possible to force pull request applied. But If you support me, I can try to create an RFC or a mail thread in internals to discuss this idea.